lib/pbench/server/utils.py

```diff
@@ -29,6 +29,29 @@ def rename_tb_link(tb, dest, logger):
         raise
 
 
+def filesize_bytes(size, logger):
```

Comment:
Not sure this works as it stands.  I wrote a little test to drive all possible combinations of input to see what would happen.  Here is the code fixed, with a few changes and a hacky test harness:

```
def filesize_bytes(size):
    size = size.strip()
    size_name = ["B", "KB", "MB", "GB", "TB"]
    try:
        parts = size.split(' ', 1)
        if len(parts) == 1:
            if size.isdigit():
                num = int(size)
                unit = 0
            else:
                for i, c in enumerate(size):
                    if not c.isdigit():
                        break
                print(f"size[:{i}] = {size[:i]}")
                num = int(size[:i])
                unit = size[i:].strip()
                print(f"num = {num}, unit = {unit}")
        else:
            num = int(parts[0])
            unit = parts[1].strip()

        if unit:
            idx = size_name.index(unit.upper())
        else:
            idx = 0
        factor = 1024 ** idx
    except Exception as exc:
        raise Exception("Invalid file size value encountered, '%s': %s", size, exc)
    else:
        return num * factor

sizes = [
    "  10  ",
]

for i in ["B", "b", "KB", "Kb", "kB", "kb", "MB", "GB", "TB"]:
    matrix = [
    f"10{i}",
    f"10 {i}",
    f"10   {i}",
    f"10{i}   ",
    f"10 {i}   ",
    ]
    sizes.extend(matrix)

for size in sizes:
    res = filesize_bytes(size)
    try:
        print(f"string = {size}, integer = {res:d}")
    except Exception as exc:
        print(f"string = {size}, exception = {exc}")
```

Reply:
There ought to be a real pytest, for sure. But you were nastier in testing this than I was. Good for you. ;-)

Reply:
Interesting, thanks for the test @portante, yeah it wasnt very robust so added a comment in our config file `# max allowed size for tarfile upload, acceptable format {X[unit] or X [unit]}`